include/itkFrequencyDomain1DImageFilter.hxx

```diff
@@ -0,0 +1,208 @@
+  typedef const typename OutputImageType::SizeType& ConstOutputSizeType;
+  ConstOutputSizeType requestedSize =
+    outputPtr->GetRequestedRegion().GetSize();
+  ConstOutputSizeType outputLargeSize =
+    outputPtr->GetLargestPossibleRegion().GetSize();
+  typedef const typename OutputImageType::IndexType& ConstOutputIndexType;
+  ConstOutputIndexType requestedIndex =
+    outputPtr->GetRequestedRegion().GetIndex();
+  ConstOutputIndexType outputLargeIndex =
+    outputPtr->GetLargestPossibleRegion().GetIndex();
+
+  typename OutputImageType::SizeType enlargedSize   = requestedSize;
+  typename OutputImageType::IndexType enlargedIndex = requestedIndex;
+  const unsigned int direction = this->GetDirection ();
+  enlargedSize[direction]  = outputLargeSize[direction];
+  enlargedIndex[direction] = outputLargeIndex[direction];
+
+  typename OutputImageType::RegionType enlargedRegion;
+  enlargedRegion.SetSize( enlargedSize );
+  enlargedRegion.SetIndex( enlargedIndex );
+  outputPtr->SetRequestedRegion( enlargedRegion );
+}
+
+
+template< typename TInputImage, typename TOutputImage >
+void
+FrequencyDomain1DImageFilter< TInputImage, TOutputImage >
+::PrintSelf( std::ostream& os, Indent indent ) const
+{
+  Superclass::PrintSelf( os, indent );
+
+  os << indent << "FilterFunction: " << m_FilterFunction << std::endl;
+}
+
+
+template< typename TInputImage, typename TOutputImage >
+const ImageRegionSplitterBase *
+FrequencyDomain1DImageFilter< TInputImage, TOutputImage >
+::GetImageRegionSplitter() const
+{
+  return this->m_ImageRegionSplitter.GetPointer();
+}
+
+
+template< typename TInputImage, typename TOutputImage >
+void
+FrequencyDomain1DImageFilter< TInputImage, TOutputImage >
+::BeforeThreadedGenerateData()
+{
+  this->m_ImageRegionSplitter->SetDirection( this->GetDirection() );
+}
+
+
+template< typename TInputImage, typename TOutputImage >
+void
+FrequencyDomain1DImageFilter< TInputImage, TOutputImage >
+::ThreadedGenerateData( const OutputImageRegionType& outputRegionForThread, ThreadIdType itkNotUsed( threadId ) )
+{
+  // get pointers to the input and output
+  const typename InputImageType * inputPtr = this->GetInput();
+  OutputImageType * outputPtr = this->GetOutput();
+
+  const typename OutputImageType::SizeType &inputSize = inputPtr->GetRequestedRegion().GetSize();
+  const unsigned int direction = this->GetDirection ();
+  const unsigned int size = inputSize[direction];
+
+  typedef ImageLinearConstIteratorWithIndex< OutputImageType > InputIteratorType;
+  typedef ImageLinearIteratorWithIndex< OutputImageType >      OutputIteratorType;
+  InputIteratorType inputIt( inputPtr, outputRegionForThread );
+  OutputIteratorType outputIt( outputPtr, outputRegionForThread );
+  inputIt.SetDirection( direction );
+  outputIt.SetDirection( direction );
+
+  // for every fft line
+  for( inputIt.GoToBegin(), outputIt.GoToBegin(); !inputIt.IsAtEnd();
+    outputIt.NextLine(), inputIt.NextLine() )
+    {
+    unsigned int i = 0;
```

Comment:
better to use `SizeValueType` instead of `unsigned int` so we can handle large images.